atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SpecificTableSweeper.java

```diff
@@ -195,19 +192,7 @@ private void processSweepResults(TableToSweep tableToSweep, SweepResults current
     }
 
     private static SweepResults getCumulativeSweepResults(TableToSweep tableToSweep, SweepResults currentIteration) {
-        long staleValuesDeleted = tableToSweep.getStaleValuesDeletedPreviously()
-                + currentIteration.getStaleValuesDeleted();
-        long cellsExamined = tableToSweep.getCellsExaminedPreviously() + currentIteration.getCellTsPairsExamined();
-        long minimumSweptTimestamp = Math.min(
-                tableToSweep.getPreviousMinimumSweptTimestamp().orElse(Long.MAX_VALUE),
-                currentIteration.getSweptTimestamp());
-
-        return SweepResults.builder()
-                .staleValuesDeleted(staleValuesDeleted)
-                .cellTsPairsExamined(cellsExamined)
-                .sweptTimestamp(minimumSweptTimestamp)
-                .nextStartRow(currentIteration.getNextStartRow())
-                .build();
+        return tableToSweep.getPreviousSweepResults().accumulateWith(currentIteration);
```

Comment:
Love it when a big chunk of code disappears like this - good indicator we've done it right.